src/util/specific.py

```diff
@@ -1,11 +1,13 @@
 # -*- coding: utf-8 -*-
 import logging
 
+from .exceptions import UpgradeError
 from .helpers import _validate_table
-from .misc import _cached
+from .misc import _cached, version_gte
 from .models import rename_model
-from .modules import rename_module
-from .pg import column_exists, rename_table, table_exists
+from .modules import odoo, rename_module
+from .orm import env
+from .pg import column_exists, format_query, parallel_execute, rename_table, table_exists
 from .report import add_to_migration_reports
```

Comment:
Import the `_get_translation_upgrade_queries` function directly from `odoo` module.
```suggestion
from .modules import rename_module
from .orm import env
from .pg import column_exists, format_query, parallel_execute, rename_table, table_exists
from .report import add_to_migration_reports

try:
    from odoo.tools.translate import _get_translation_upgrade_queries
except ImportError:
    if version_gte("16.0"):
        raise
```

Reply:
I don't think it's a good idea. This function is only needed for 16.0+, so raising an `ImportError` in lower versions that import this file would be harmful.

With the current implementation, if some of those older versions happen to use the new `translation2jsonb` function, only then they'd fail. That's better IMHO.

Reply:
It's still the case. It only reraises the `ImportError` if the version is *older* than 16.

And this is exactly what I want to catch. If the `_get_translation_upgrade_queries` is removed[^1] from odoo (like it nearly happen in [I can't find back the odoo PR]), I want the CI to directly fail (the new functions are not covered by the CI).


[^1]: or renamed or moved.

Reply:
Yikes, true!